### PR TITLE
[admin] 404 on Device change page raises 500 internal server error

### LINKED ISSUE #165: [admin] 404 on Device change page raises 500 internal server error
How to replicate:

- open device admin list
- open a device change page with the UUID modified so to trigger a 404

Expected result: 404

Actual result: 500 internal server error.

A test to ensure we won't have regressions will be needed.

### PR DESCRIPTION
**Pull request by nepython**
close #165

### CODE REVIEW

**Comment by nepython** [comment]
@nemesisdesign, can you please review this. I have tested it thoroughly to make sure minimum to no changes being suggested. :grinning:

**Comment by nepython** [comment]
> "A test to ensure we won't have regressions will be needed."

@nemesisdesign, it's done, I have tested it. Prior to the change, the test is failing due to the bug.
Also, added doc-strings, to explain everything.  I have used try-except, as despite many workarounds that I tried out **DoesNotExist** exception was being raised.
:smile:

File: django_netjsonconfig/tests/test_admin.py [hunk 1]
```diff
@@ -255,6 +256,22 @@ def test_preview_template(self):
...
+        d = Device(name='test')
+        path = reverse('admin:django_netjsonconfig_device_change', args=[d.pk])
+        """
+            If views are rendered, a does not exist exception shall be raised
+            leading to internal server error Http response 500.
+        """
+        try:
+            response = self.client.get(path)
+        except Device.DoesNotExist:
+            response = HttpResponse(status=500)
```
Comment (nemesisdesign) [inline django_netjsonconfig/tests/test_admin.py]: this should be removed, only the line `response = self.client.get(path)` should be kept.
If the exception is raised, the test will fail with an error and that's fine.

File: django_netjsonconfig/tests/test_admin.py [hunk 2]
```diff
@@ -255,6 +256,22 @@ def test_preview_template(self):
...
+    def test_wrong_uuid_field_in_change_404(self):
+        """
+            Device not saved in the databse
+        """
+        d = Device(name='test')
+        path = reverse('admin:django_netjsonconfig_device_change', args=[d.pk])
+        """
+            If views are rendered, a does not exist exception shall be raised
+            leading to internal server error Http response 500.
+        """
```
Comment (nemesisdesign) [inline django_netjsonconfig/tests/test_admin.py]: since this test needs just two lines of code, this dosctring is really overkill, please remove it

**Comment by nepython** [comment]
I have made the changes, it's ready for review :smiley:

File: django_netjsonconfig/tests/test_admin.py [hunk 3]
```diff
@@ -255,6 +255,15 @@ def test_preview_template(self):
         self.assertNotContains(response, 'system')
         self.assertNotContains(response, 'hostname')
 
+    def test_wrong_uuid_field_in_change_404(self):
+        """
+            Device not saved in the databse
+        """
```
Comment (nemesisdesign) [inline django_netjsonconfig/tests/test_admin.py]: again I have to ask you to please remove the docstring, it means remove the entire docstring.

Comment (nepython) [reply django_netjsonconfig/tests/test_admin.py]: @nemesisdesign, I was going to do so but you had explicitly mentioned `**this** docstring` in your comment. Hence, left the above one. Nevertheless, won't include any docstrings in further PRs unless a different logic is being used :smiley:

Comment (nemesisdesign) [reply django_netjsonconfig/tests/test_admin.py]: as a general rule, follow the conventions used in the files you modify and add comments whenever the code is not easy to read and understand at glance

**Pull request merged**